### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by KyleRickards** [saschaludwig/OnAirScreen#2]
Hi

Is it possible to use the clock purely as a digital display please? (Without linking to a Raspberry Pi)

I use Ubuntu and would quite like to have this on my desktop at all times so I can see where I am up to in a show?

Many thanks
Kyle

**Comment by saschaludwig** [comment]
Hi Kyle

It is written in PyQT4 and can be run on windows, mac and of course linux. You need python, qt and the pyqt bindings if you want to run it from source. So running it on your ubuntu system should be possible.
It can run in normal window mode or in full screen.

Best regards
Sascha

**Comment by KyleRickards** [comment]
Hi Sascha

Many thanks for this.

I have python and things but I’m not sure how to build from source? Are you able to help with this?

Many many thanks for replying though!

Kyle

**Comment by KyleRickards** [comment]
Hi 

I have installed 

pyqt4 
pyqt-dev-tools
python-qt4-dbg

but still cannot get the program to run? I am getting this error when i run ./start

Traceback (most recent call last):
  File "./start.py", line 46, in <module>
    from mainscreen import Ui_MainScreen
ImportError: No module named mainscreen

I do not know what else I need to get? Is there anything I can try further please as this is exactly what I have been after.

Kyle

**Comment by KyleRickards** [comment]
Edited to add - I have run 'make' and am now getting this error?

Traceback (most recent call last):
  File "./start.py", line 48, in <module>
    import ntplib
ImportError: No module named ntplib

**Comment by KyleRickards** [comment]
![image](https://user-images.githubusercontent.com/21976755/40586751-5fdcb694-61be-11e8-97ed-b43413b25827.png)

**Comment by KyleRickards** [comment]
I still don't really understand how to set the ntp or change the language from Deutsch but thank you, this looks very good. (I don't really understand a lot of computers or Python)

**Comment by saschaludwig** [comment]
As you already found out: you have to run "make" so it can create the python UI files. :)

The "Clock not NTP syncronized" error shows when your local clock differs too much from the ptbtime1.ptb.de server or when no ntp server is set to compare the local clock to. You can set an NTP server in the settings. See the readme.md for keyboard shortcuts.

There is no language setting right now. You need to change the code if you want it in another language. I don't have a decent idea how to manage multiple languages in an easy manner. But it is something I'm thinking about.

If you have any other questions feel free to ask :)

Best wishes Sascha

**Comment by KyleRickards** [comment]
Many thanks for all the help Sascha - it's very much appreciated :)

Kyle

**Comment by st1nga** [comment]
I have changed start1 to be...

`    def updateBacktimingText(self):
        now = datetime.now()
        hour = now.hour
        minute = now.minute
        second = now.second
        remain_min = 60-minute
        if hour > 12:
            hour -= 12

        if minute == 0:
            string = "its %d o'clock" % hour
        if (minute > 0 and minute < 15) or (minute >= 16 and minute <= 29):
            string = "its %d minute%s past %d" % (minute, 's' if minute>1 else '', hour)
        if minute == 15:
            string = "its a quarter past %d" % (hour)
        if minute == 30:
            string = "its half past %d" % (hour)
        if minute == 45:
            string = "its a quarter to %d" % (hour)
        if (minute >= 31 and minute <= 44) or (minute >= 46 and minute <= 59):
            string = "its %d minute%s to %d" % (remain_min, 's' if remain_min>1 else '', 1 if hour == 12 else hour+1)
`

**Comment by KyleRickards** [comment]
Thank you! Can I just copy that code in to the script?
--------------------------------------------
On Thu, 7/6/18, st1nga <notifications@github.com> wrote:

 Subject: Re: [saschaludwig/OnAirScreen] Question (#2)
 To: "saschaludwig/OnAirScreen" <OnAirScreen@noreply.github.com>
 Cc: "KyleRickards" <kyle.rickards1@btinternet.com>, "Author" <author@noreply.github.com>
 Date: Thursday, 7 June, 2018, 13:30
 
 I have changed start1 to
 be...
 `    def updateBacktimingText(self):
 
 now = datetime.now()
 
 hour = now.hour
 
 minute = now.minute
 
 second = now.second
 
 remain_min = 60-minute
 
 if hour > 12:
 
 hour -= 12
     if minute == 0:
         string = "its %d o'clock" % hour
     if (minute > 0 and minute < 15) or (minute >=
 16 and minute <= 29):
         string = "its %d minute%s past %d" %
 (minute, 's' if minute>1 else '', hour)
     if minute == 15:
         string = "its a quarter past %d" % (hour)
     if minute == 30:
         string = "its half past %d" % (hour)
     if minute == 45:
         string = "its a quarter to %d" % (hour)
     if (minute >= 31 and minute <= 44) or (minute
 >= 46 and minute <= 59):
         string = "its %d minute%s to %d" %
 (remain_min, 's' if remain_min>1 else '',
 1 if hour == 12 else hour+1)
 
 `
 
 —
 You are
 receiving this because you authored the thread.
 Reply
 to this email directly, view
 it on GitHub, or mute
 the thread.

**Comment by KyleRickards** [comment]
Hi Sascha

Where do I copy this code into please?
Thanks 
Kyle

**Comment by KyleRickards** [comment]
Many thanks for all the help Sascha - it's very much appreciated :)

Kyle

**Comment by saschaludwig** [comment]
You have to replace the corresponding code-block with the german text strings in it. Like the pasted code says. The function is "updateBacktimingText"

**Comment by saschaludwig** [comment]
I'm about to implement a simple language switcher for DE/US in the upcoming weeks.
I will also add a optional "seconds" display below the hh:mm as one of the users has suggested.

**Comment by KyleRickards** [comment]
Ah should I wait before making the change?
--------------------------------------------
On Wed, 20/6/18, Sascha Ludwig <notifications@github.com> wrote:

 Subject: Re: [saschaludwig/OnAirScreen] Question (#2)
 To: "saschaludwig/OnAirScreen" <OnAirScreen@noreply.github.com>
 Cc: "KyleRickards" <kyle.rickards1@btinternet.com>, "Author" <author@noreply.github.com>
 Date: Wednesday, 20 June, 2018, 21:03
 
 I'm about to implement a
 simple language switcher for DE/US in the upcoming weeks.
 
 I will also add a optional "seconds" display below
 the hh:mm as one of the users has suggested.
 
 —
 You are receiving this because you authored the thread.
 Reply to this email directly, view
 it on GitHub, or mute
 the thread.

**Comment by saschaludwig** [comment]
I implemented a language switcher for the textclock and added a customizable date format in the python3_pyQt5 branch

**Comment by KyleRickards** [comment]
Oh thank you, I will try this later

Sent from my iPad

On 5 Feb 2019, at 14:00, Sascha Ludwig <notifications@github.com> wrote:

I implemented a language switcher for the textclock and added a customizable date format in the python3_pyQt5 branch

**Comment by KyleRickards** [comment]
Do I need to download and run it again?

Sent from my iPad

On 5 Feb 2019, at 14:00, Sascha Ludwig <notifications@github.com> wrote:

I implemented a language switcher for the textclock and added a customizable date format in the python3_pyQt5 branch

**Comment by KyleRickards** [comment]
Hi
I cannot get it working at all now, I have listed the error messages I am getting on the comments page - are you able to help any futher?
Kyle

    On Tuesday, 5 February 2019, 14:00:08 GMT, Sascha Ludwig <notifications@github.com> wrote:  
 
 
I implemented a language switcher for the textclock and added a customizable date format in the python3_pyQt5 branch

**Comment by KyleRickards** [comment]
Hiya, 

So this is where I got to last night:

I cannot get virtual environment to work with different python version however it’s not an issue as I don’t mind rebuilding the OS if needed. 

So as per other email I did get the newest version working in non virtual environment. I changed the logo and disabled ntp check and now it won’t load again with the error messages I think I sent last night (but I can resend?)

I have redownloaded the beta and it makes the file but gives me errors when trying to run.

Interestingly I finally got the master version (the early version, not in virtual environment) working on Linux Mint 19.1 BUT it loaded up with my radio slogan Reclaimed Radio) so it’s saving the information somewhere but I don’t know where or how it’s doing it? (This is after a complete reinstall/build)

Whew! Hope that made sense!


Sent from my iPad
